### PR TITLE
feat: implement discard modal

### PR DESCRIPTION
**Pull request by nsilva-spryker**
Close [HRZ-2921](https://spryker.atlassian.net/browse/HRZ-2921)

[HRZ-2921]: https://spryker.atlassian.net/browse/HRZ-2921?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { ButtonType } from '@spryker-oryx/ui/button';
+import { i18n, Size } from '@spryker-oryx/utilities';
+import { html, LitElement, TemplateResult } from 'lit';
+import { property } from 'lit/decorators.js';
+import { discardModalStyles } from './discard-modal.styles';
+
+export class DiscardModalComponent extends LitElement {
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.ts]: Discard what? Probably better to name it `DiscardPickingComponent`

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('passes the a11y audit', async () => {
+    await expect(element).shadowDom.to.be.accessible();
+  });
+
+  it('should render modal', () => {
+    expect(element.renderRoot.querySelector('oryx-modal')).not.toBeNull();
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts]: expect(element).toContainElement('oryx-modal')

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  describe('when modal is opened', () => {
+    beforeEach(() => {
+      element.open = true;
+    });
+
+    it('modal should have opened attribute', () => {
+      expect(
+        element.renderRoot.querySelector('oryx-modal')?.hasAttribute('open')
+      ).toBe(true);
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts]: expect(element).toContainElement('oryx-modal[open]')

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    });
+
+    describe('and when modal is closed', () => {
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts]: `    describe('and modal is closed', () => {`

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      beforeEach(() => {
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.spec.ts:42]:
```
describe('and modal is closed', () => {
    beforeEach(() => {
      element.renderRoot.querySelector('oryx-modal')?.dispatchEvent(new CustomEvent('oryx.close'));
    });
  
    it('should close the modal', () => {
      expect(element).not.toContainElement('oryx-modal[open]')
    });
  });

  describe('and close button is clicked', () => {
    beforeEach(() => {
      element.renderRoot.querySelector('button')?.dispatchEvent(new MouseEvent('click'));
    });
  
    it('should close the modal', () => {
      expect(element).not.toContainElement('oryx-modal[open]')
    });
  });
```

File: libs/domain/picking/src/components/discard-modal/discard-modal.def.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { componentDef } from '@spryker-oryx/core';
+
+export const discardModalComponent = componentDef({
+  name: 'oryx-discard-modal',
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.def.ts]: oryx-discard-picking

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  @property({ type: Boolean }) open?: boolean;
+
+  protected override render(): TemplateResult {
+    return html`
+      <oryx-modal
+        ?open=${this.open}
+        enableFooter
+        preventCloseByBackdrop
+        footerButtonFullWidth
+        @oryx.close=${this.close}
```
Comment (Sinucid) [inline libs/domain/picking/src/components/discard-modal/discard-modal.component.ts]: I'm wondering, should this modal close on esc? Are we preventing closing on backdrop, but not on esc keypress? Looks like both ways should be prevented

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/picking/src/components/discard-modal/discard-modal.component.ts (hunk 7 above)
Comment (nsilva-spryker) [reply libs/domain/picking/src/components/discard-modal/discard-modal.component.ts]: Yup, both ways should be prevented. `preventCloseByEscape` added

Comment (Sinucid) [reply libs/domain/picking/src/components/discard-modal/discard-modal.component.ts]: Then this close handler no more relevant, you can drop it and related tests

**Review by Sinucid — CHANGES_REQUESTED** [review]

**Review by Sinucid — APPROVED** [review]

**Pull request merged**